crates/core/src/models/mod.rs

```diff
@@ -402,3 +412,76 @@ impl ConvertibleStarknetTransaction for StarknetTransaction {
         })
     }
 }
+
+pub struct StarknetEvent(Event);
+
+impl StarknetEvent {
+    pub fn new(sn_event: Event) -> Self {
+        Self(sn_event)
+    }
+}
+
+#[async_trait]
+impl ConvertibleStarknetEvent for StarknetEvent {
+    async fn to_eth_log(
+        &self,
+        client: &dyn KakarotClient,
+        block_hash: Option<H256>,
+        block_number: Option<U256>,
+        transaction_hash: Option<H256>,
+        log_index: Option<U256>,
+        transaction_index: Option<U256>,
+    ) -> Result<Log, EthApiError> {
+        // If event `from_address` does not equal kakarot address, return early
+        if self.0.from_address != client.kakarot_address() {
+            return Err(EthApiError::OtherError(anyhow::anyhow!("Kakarot Filter: Event is not part of Kakarot")));
+        }
+
+        // Derive the evm address from the last item in the `event.keys` vector and remove it
+        let (last_key_as_addr, keys_without_address) =
+            self.0.keys.split_last().ok_or_else(|| {
+                EthApiError::OtherError(anyhow::anyhow!("Kakarot Filter: Event is not part of Kakarot"))
```

Comment:
I think the error message is a bit misleading.
If we pass the verification of `self.0.from_address != client.kakarot_address()`, then it must be a kakarot event. 

But it can be an event that's not related to deploying evm smart contract

Reply:
it could also just be a starknet event that doesn't parse as an evm event, no?